### PR TITLE
Sanitize Password Length

### PR DESCRIPTION
**Pull request by vincentsalucci**
## Code Changes
- **PasswordGenerationService.cs**: Created `SanitizePasswordLength` method that can be called with UI changes or for password generation only.  Maintains existing workflows but will now properly update the UI if a minimum length condition is met.

### CODE REVIEW

File: src/Core/Services/PasswordGenerationService.cs [hunk 1]
```diff
@@ -691,5 +648,56 @@ private async Task AppendRandomNumberToRandomWordAsync(List<string> wordList)
...
+        private static void SanitizePasswordLength(PasswordGenerationOptions options, bool forGeneration)
+        {
+            var minUppercaseCalc = 0;
+            var minLowercaseCalc = 0;
+            var minNumberCalc = options.MinNumber;
+            var minSpecialCalc = options.MinNumber;
+
+            if (options.Uppercase.GetValueOrDefault() && options.MinUppercase.GetValueOrDefault() <= 0) {
+                minUppercaseCalc = 1;
+            } else if (!options.Uppercase.GetValueOrDefault()) {
```
Comment (kspearrin) [inline src/Core/Services/PasswordGenerationService.cs]: formatting on c# is different than JS. should be new line.

File: src/Core/Services/PasswordGenerationService.cs [hunk 2]
```diff
@@ -691,5 +648,56 @@ private async Task AppendRandomNumberToRandomWordAsync(List<string> wordList)
...
         }
+
+        private static void SanitizePasswordLength(PasswordGenerationOptions options, bool forGeneration)
+        {
+            var minUppercaseCalc = 0;
+            var minLowercaseCalc = 0;
+            var minNumberCalc = options.MinNumber;
+            var minSpecialCalc = options.MinNumber;
+
+            if (options.Uppercase.GetValueOrDefault() && options.MinUppercase.GetValueOrDefault() <= 0) {
```
Comment (kspearrin) [inline src/Core/Services/PasswordGenerationService.cs]: formatting on c# is different than JS. should be no whitespace between `if(`

**Comment by vincentsalucci** [comment]
@kspearrin - IDE suggested making the `SanitizePasswordLength` method `static`.  Did some research and didn't see any reason not to do it. What are your thoughts on that and when to make a method static?

File: src/Core/Services/PasswordGenerationService.cs [hunk 3]
```diff
@@ -691,5 +648,67 @@ private async Task AppendRandomNumberToRandomWordAsync(List<string> wordList)
...
+        {
+            var minUppercaseCalc = 0;
+            var minLowercaseCalc = 0;
+            var minNumberCalc = options.MinNumber;
+            var minSpecialCalc = options.MinNumber;
+
+            if(options.Uppercase.GetValueOrDefault() && options.MinUppercase.GetValueOrDefault() <= 0)
+            {
+                minUppercaseCalc = 1;
+            } else if(!options.Uppercase.GetValueOrDefault())
```
Comment (kspearrin) [inline src/Core/Services/PasswordGenerationService.cs]: else if should be on a new line. Is formatter not fixing that?

File: src/Core/Services/PasswordGenerationService.cs [hunk 4]
```diff
@@ -691,5 +648,67 @@ private async Task AppendRandomNumberToRandomWordAsync(List<string> wordList)
             var num = await _cryptoService.RandomNumberAsync(0, 9);
             wordList[index] = wordList[index] + num;
         }
+
+        private static void SanitizePasswordLength(PasswordGenerationOptions options, bool forGeneration)
```
Comment (kspearrin) [inline src/Core/Services/PasswordGenerationService.cs]: let's not make it static. we havent been doing that with others.

**Pull request merged**